### DISCUSSION TITLE
Ruin scan mission is plagued with issues

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#13740] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

1. **The mission fails randomly between scanning all points of interest and reaching the end of the level.** (May've been fixed when the requirement of bringing the scanners to the outpost was removed?)

This is a rare failure which I couldn't intentionally reproduce in the 7 attempts I've done yesterday. Happens both in singleplayer and in multiplayer, reported by multiple people. Happened to me about 4 times across 3 full campaigns, which includes two unmodded campaigns and one modded one.
To people commenting under this discussion, please share any information you have on this.

2. **Ruin scanners can be placed inside solid walls, but you can't get them back without breaking the wall.**

This one is funny. Your character's hand can reach deep enough inside the solid wall of the ruin to place the scanner there. Happens accidentally sometimes. To get it back you have to to cut the wall segment open. Which you can't really do with the walls of the artifact room. Maybe placing the scanners in walls shouldn't be possible to begin with.

3. **The texts don't make it clear whether you have to bring the scanners with data back or not, or how many scanners you have to use.** (In reality you can use any amount of scanners and do whatever you want with them afterwards.)

Mission description: _To further their research, a group of scientists in [location1] have asked you to scan the nearby ruin. Venture into the ruin and scan the marked rooms using the provided scanners._
All scans done message: _**Mission accomplished.** Ruins are successfully scanned._
Round end success message: _The ruins were successfully scanned and **the data was delivered**._
  
The final success message sounds like we brought the scanners back. A lot of people question the intended way of completing the mission upon reading its description, and due to the variety of ways in which it can be completed, the answers they get from other players tend to vary.
690a9e3ec1: "Scan ruins mission: remove the requirement for bringing back the scanners." suggests that the mission success message simply hasn't been updated, but the description of the mission itself is still vague and can cause confusion.

### Reproduction steps

Refer to the section above.
Here are the results of my own multiplayer tests. All of these resulted in successful completion.

Attempt 1. Used 3 scanners. Stored in a cabinet on the submarine.
Attempt 2. Same as the first one.
Attempt 3. Used 3 scanners. Stored them in players' inventories.
Attempt 4. Used 1 scanner for all points of interest. Stored it in a cabinet.
Attempt 5. Used 1 scanner for all points. Stored in inventory.
Attempt 6. Used 3 scanners. Left them in ruins.
Attempt 7. Used 2 scanners, lost one inside a wall. Left them all in ruins.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.3.0.2

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Thanks for the feedback!

> The mission fails randomly between scanning all points of interest and reaching the end of the level. (May've been fixed when the requirement of bringing the scanners to the outpost was removed?)

The requirement was removed on last summer. Have you still been encountering issues with this?

> Ruin scanners can be placed inside solid walls, but you can't get them back without breaking the wall.

This was an old bug, but it should have been fixed. Have you still been encountering the issue?

> The texts don't make it clear whether you have to bring the scanners with data back or not, or how many scanners you have to use. (In reality you can use any amount of scanners and do whatever you want with them afterwards.)

Probably because you previously had to take the scanners back, a condition that we've now removed. We edited the texts a bit, but maybe not enough.

**Reply by NotWendy** [reply]
> (On random mission failure) The requirement was removed on last summer. Have you still been encountering issues with this?

I'm honestly not sure. It's possible that this bug no longer exists. Since my memory on when I last encountered it is hazy, I was hoping anybody else would have info on this. But so far, others who have seen it have been as unsure about the date when it last happened as I am.

> (On scanners in walls) This was an old bug, but it should have been fixed. Have you still been encountering the issue?

Yes, this one absolutely does happen in the current stable version, where I did my runs. Unless it was fixed after the last stable release, in Unstable, it's still a very real problem. I had it happen twice (by accident) throughout the seven mission mode runs, so it should be easy to reproduce intentionally.
